library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt

```diff
@@ -37,58 +37,53 @@ class LocationCredentialsProvider {
      */
     constructor(context: Context, identityPoolId: String, region: AwsRegions) {
         this.context = context
-        securePreferences = EncryptedSharedPreferences(context, PREFS_NAME)
-        securePreferences.initEncryptedSharedPreferences()
-        securePreferences.put(METHOD, "cognito")
-        securePreferences.put(IDENTITY_POOL_ID, identityPoolId)
-        securePreferences.put(REGION, region.regionName)
+        initPreference(context)
+        securePreferences?.put(METHOD, "cognito")
+        securePreferences?.put(IDENTITY_POOL_ID, identityPoolId)
+        securePreferences?.put(REGION, region.regionName)
     }
 
     /**
-     * Initializes with an API key.
+     * Initializes with region.
      * @param context The application context.
-     * @param apiKey The API key for authentication.
+     * @param region The region for Cognito authentication.
      */
-    constructor(context: Context, apiKey: String) {
+    constructor(context: Context, region: AwsRegions) {
```

Comment:
If this constructor is used just for the custom Credential provider method, might as well pass it here rather than using the `initializeLocationClient` method. Wdyt?

Reply:
`initializeLocationClient` is a suspend function, so instead of initializing the coroutine in the constructor of `LocationCredentialsProvider`, we can directly call the function here.

Reply:
I think this is related to my other open comment, as I don't think we need the extra translation layer that is done in `setCustomCredentials` between `Credentials(aws.smithy.kotlin.runtime.auth.awscredentials)` and `Credentials( aws.sdk.kotlin.services.cognitoidentity.model.Credentials)`. If we don't invoke the the `credentials.resolve()` method and just assign the credential porvider directly to the `LocationClient`, then we could pass in the credential provider directly in the constructor as there is no async behavior that needs to occur at instantiation. Thus the constructor could look something like this:

```kotlin
    constructor(context: Context, region: AwsRegions, provider: CredentialsProvider) {
        this.context = context
        this.credentialsProvider = provider
        securePreferences = initPreference(context)
        securePreferences.put(METHOD, "custom")
        securePreferences.put(REGION, region.regionName)
    }
```

Which then allows the caller to call `getLocationClient` and get a properly configured client. Wdyt?

Reply:
As discussed internally, can be addressed as fast follow.